src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -3081,6 +3073,9 @@ self =>
       val parentPos = o2p(in.offset)
       val tstart1 = if (body.isEmpty && in.lastOffset < tstart) in.lastOffset else tstart
 
+      // we can't easily check this later, because `gen.mkParents` adds the default AnyRef parent, and we need to warn based on what the user wrote
+      if (name == nme.PACKAGEkw && parents.nonEmpty) deprecationWarning(tstart, s"package object inheritance is deprecated", "2.13.0")
+
```

Comment:
It's a pity it's quite pithy. I wonder if it should say another word, like, "And we're taking package objects away from you entirely in the next release." First they took away my inheritance, then my objects, next there will be no packages at all under the tree. No fun living at the orphanage.

Reply:
I blame the boomers. Anyway, I added some links and verbiage.